data/housenumber-filters-spanyolret.yaml

```diff
@@ -15,4 +15,9 @@ filters:
   felsohatar_ut:
     ranges:
     - {end: '36', isOdd: 'false', start: '2'}
+refstreets:
+  # 'OSM Name 1': 'Ref Name 1'
+  # 'OSM Name 2': 'Ref Name 2'
+'Felsőhatár utca': 'Felső határ út'
+'Felső határút': 'Felső határ út'
```

Comment:
21-22. sor elé tegyél kérlek 2 space-t, különben nem fog a refstreets alá tartozni.

@bkil próbáltam erre sémát írni, hogy ha van refstreets akkor az egy str->str dictionary, de nem jöttem rá a módjára. Ha van időd és kedved, ránézhetnél. (Ebben az esetben esetleg elkapná a problémát ha azt is megköveteled, hogy nem lehet üres a dictionary.)

Reply:
Nem-ürességi megszorítást azt hiszem támogat. Különben nem vagyok megelégedve a validátorral. Egyrészt az egyik yaml fájlt sem tudja hatékonyan ellenőrizni, másrészt pedig csak megengedő vizsgálatok vannak benne - kizáróak nem. Pedig egy valódi validátor támogatja a zárt világ ellenőrzést. Van egy másik tool is amit library-ként is lehet használni. Vagy fapadosan kézzel is parszolhatjuk a yaml-ot ellenőrizni. Esetleg JSON vagy XML-re is hozhatjuk - azokra már több validátor létezik. Ez a kérdés annyiban érdekel, hogy szeretnék a jövőben josm presetet is egy barátságos nyelvből generálni (akár YAML vagy HOCON), és hasonló validátorokra ott is szükségem lesz.

Reply:
beraktam a space-eket. 
én ettől a yaml-tól nem vagyok hasra esve... állandóan küzdök ezekkel a space-ekkel, nekem jobban bejön a json. csak akkor a kommentek...
kedvencem amikor a komment végén lévő trailing space-ért szól a validátor

Reply:
Nos, bár a maximalista énünk szereti a trailing space-ek mellőzését, ezt lekapcsolhatjuk a validátorból, mert kevesebb bajt csinál. A behúzásokkal könnyebb mellé lőni.

Én pedig jobban szeretem a kevesebb boiler plate-et és behúzásokat és a JSONT kicsit alacsony szintűnek tartom, de persze a usereink igényeire kellene optimalizálni akik ezeket ténylegesen szerkesztik. Van jobb alternatíva azon kívül, hogy csinálunk a szerkesztésére egy web formot?
https://www.zionandzion.com/json-vs-xml-vs-toml-vs-cson-vs-yaml/
https://stackoverflow.com/questions/8992262/human-editable-json-like-or-yaml-like-program-configuration-in-java
https://github.com/jclawson/jackson-dataformat-hocon
https://news.ycombinator.com/item?id=14315135

Esetleg kapcsoljuk vissza a json parszolás lehetőségét...?

Reply:
Jó lesz ez, csak morgok. Egyelőre maradjon így. 
Kellene egy rendes validátor localban ,vagy webes, akkor nem kellene köröket futnom feltöltés után, persze nem óriási gondok ezek.

Reply:
Lokálban a validátor: `make check-filters` ami hamar lefut, vagy ha csak a séma kell akkor `make check-filters-schema`. Ehhez minimális függőség kell (python, easy_install/pip, sémához pip install yamale, szintaxishoz yamllint).

Reply:
Ha nincs Make-ed, kézzel is lehet egyetlen fájlra futtatni: `yamllint data/housenumber-filters-dobogo.yaml`. Vagy másik mappából: `yamllint -c ../.yamllint housenumber-filters-dobogo.yaml`, illetve pedánsoknak ugyanez `-c ../.yamllint.strict` configgal vagy `make check-full-filters`.

Reply:
így megy:

`yamllint data/housenumber-filters-dobogo.yaml -c .yamllint.strict`

make nem megy windowson?

Reply:
Tud menni a make is, de először azt is fel kell tenni. Nem tudom biztosan, de @vmiklos ezt javasolta szerintem: https://www.cygwin.com/install.html
De teljesen jó így is parancs amit írtál. Szerintem tud minden fájlt is csekkolni make nélkül ha kell, próbáld ki:
`yamllint -c .yamllint data/*.yaml` (a strict-hez a többi fájlt még először helyre kell pofozni, de új fájlnál bátran használd azt)

Reply:
A séma ellenőrzést is meg tudod hívni kézzel: `yamale -s data/housenumber-filters.schema.yaml data/housenumber-filters-dobogo.yaml`, illetve a többi ellenőrzést is kilesheted a Makefile-ból ha érdekel.

Reply:
sima make-t felraktam de ott baja avan a $$F-el
*.yaml-t nem ette meg. cygwint nem raktam fel egy for ciklus miatt, mert megoldottam így:

`for %%i in (data/housenumber-filters-*.yaml) do yamllint data/%%i -c .yamllint`